docs/contributing/react-best-practices/18-anti-overengineering.md

```diff
@@ -0,0 +1,247 @@
+# Anti-overengineering: Add Tools Only with Clear Signals
+
+Default to React primitives. Add tools when they remove real complexity, not to match trends.
+
+**Practice**
+- Start with built-in React patterns (useState, useContext, custom hooks)
+- Add external libraries only when you have **concrete signals** they solve a real pain point
+- Don't add tools preemptively "in case we need them later"
+- Evaluate libraries based on the problem they solve, not popularity
```

Comment:
There's some missed nuance here. Generally, totally agree that one should value how well it solves the problem over how popular it is; however, popularity is can be a (admittedly imperfect) proxy for risk of abandonment (though I suppose it depends on how 'popularity' is defined).